Pod/Classes/XExtensionItem.h

```diff
@@ -142,37 +122,40 @@ typedef UIImage *(^XExtensionItemThumbnailProvidingBlock)(CGSize suggestedSize);
- *
- *  @param itemBlock    (Required) A block that will be called if the activity with the specified type is selected. It
- *  should return the item to be provided to this activity.
- *  @param activityType (Required) The activity type that the subject should be provided to.
+ An optional string describing the item content. In the case of social activities, for example, this text could be
+ shared alongside a URL.
+ 
+ @see `NSExtensionItem`
  */
-- (void)registerItemProvidingBlock:(XExtensionItemProvidingBlock)itemBlock forActivityType:(NSString *)activityType;
+@property (nonatomic, copy) NSAttributedString *attributedContentText;
+- (void)setAttributedContentText:(NSAttributedString *)attributedContentText forActivityType:(NSString *)activityType;
 
 /**
- *  Registers a subject for activities that support separate subject and data fields.
- *
- *  @param subject      (Required) A string to use as the contents of the subject field.
- *  @param activityType (Required) The activity type that the subject should be provided to.
+ An array of additional media associated with the extension item. These items must be of type `NSString`, `NSURL`,
+ `UIImage`, or `NSItemProvider` and will be passed to the selected activity/extension. For example, you could add
+ an image attachment to be shared alongside a URL.
  */
-- (void)registerSubject:(NSString *)subject forActivityType:(NSString *)activityType;
+@property (nonatomic, copy) NSArray *additionalAttachments;
+- (void)setAdditionalAttachments:(NSArray *)attachments forActivityType:(NSString *)activityType;
 
 /**
- *  Registers a thumbnail preview image for activities that support a preview image.
+ *  Sets a thumbnail preview image for activities that support a preview image.
  *
- *  @param thumbnailBlock    (Required) A block that will be called with the suggested size for the thumbnail image, in
- *  points. It should return an image using the appropriate scale for the screen. Images provided at the suggested size 
- *  will result in the best experience.
- *  @param activityType      (Required) The activity type that the subject should be provided to.
+ *  @param thumbnailProvider  (Required) A block that will be called with the suggested size for the thumbnail image, in
+ *  points, and the activity type chosen by the user. It should return an image using the appropriate scale for the
+ *  screen. Images provided at the suggested size will result in the best experience.
  */
-- (void)registerThumbnailProvidingBlock:(XExtensionItemThumbnailProvidingBlock)thumbnailBlock forActivityType:(NSString *)activityType;
+- (void)setThumbnailProvider:(XExtensionItemThumbnailProvidingBlock)thumbnailProvider;
+
+@end
+
+@interface XExtensionItemSource (ProviderBlockInitializers)
+
+- (instancetype)initWithURLProvider:(NSURL *(^)(NSString *activityType))urlProvider;
+- (instancetype)initWithFileURLProvider:(NSURL *(^)(NSString *activityType))fileURL ofType:(NSString *)typeIdentifier;
```

Comment:
I would make the selector `initWithFileURLProvider:typeIdentifier` to match `NSItemProvider`